x-pack/platform/plugins/shared/lens/public/visualizations/datatable/components/toolbar.tsx

```diff
@@ -17,41 +17,71 @@ import { DEFAULT_PAGE_SIZE } from './table_basic';
 import {
   DEFAULT_HEADER_ROW_HEIGHT,
   DEFAULT_HEADER_ROW_HEIGHT_LINES,
+  DEFAULT_ROW_HEIGHT,
   DEFAULT_ROW_HEIGHT_LINES,
+  ROWS_HEIGHT_OPTIONS,
+  ROW_HEIGHT_LINES_KEYS,
 } from './constants';
 
+type LineCounts = {
+  [key in keyof typeof ROW_HEIGHT_LINES_KEYS]: number;
+};
+
 export function DataTableToolbar(props: VisualizationToolbarProps<DatatableVisualizationState>) {
   const { state, setState } = props;
+
+  const [lineCounts, setLineCounts] = useState<LineCounts>({
+    [ROW_HEIGHT_LINES_KEYS.headerRowHeightLines]:
+      state.headerRowHeightLines && state.headerRowHeightLines > 0
+        ? state.headerRowHeightLines
+        : DEFAULT_HEADER_ROW_HEIGHT_LINES,
+    [ROW_HEIGHT_LINES_KEYS.rowHeightLines]:
+      state.rowHeightLines && state.rowHeightLines > 0
+        ? state.rowHeightLines
+        : DEFAULT_ROW_HEIGHT_LINES,
+  });
+
   const onChangeHeight = useCallback(
     (
       newHeightMode: RowHeightMode | undefined,
       heightProperty: string,
-      heightLinesProperty: string,
-      defaultRowHeight = DEFAULT_ROW_HEIGHT_LINES
+      heightLinesProperty: keyof typeof ROW_HEIGHT_LINES_KEYS
     ) => {
-      const rowHeightLines =
-        newHeightMode === RowHeightMode.single
-          ? 1
-          : newHeightMode !== RowHeightMode.auto
-          ? defaultRowHeight
-          : undefined;
+      let newRowHeightLines;
+
+      switch (newHeightMode) {
+        case RowHeightMode.auto:
+          newRowHeightLines = ROWS_HEIGHT_OPTIONS.auto;
+          break;
+        case RowHeightMode.single:
+          newRowHeightLines = ROWS_HEIGHT_OPTIONS.single;
+          break;
+        default:
+          newRowHeightLines = lineCounts[heightLinesProperty];
+      }
```

Comment:
I would rather prefer a `switch` to be the only expression in a scope.
In this case I would like either something explicit of multiple `if`s or even a nested ternary like the original one.
```suggestion
    let newRowHeightLines = lineCounts[heightLinesProperty];
    if( newHeightMode === RowHeightMode.auto);
      newRowHeightLines = ROWS_HEIGHT_OPTIONS.auto;
    }
    if( newHeightMode === RowHeightMode.single){
      newRowHeightLines = ROWS_HEIGHT_OPTIONS.single;
    }
```

Reply:
No problem! I will change accordingly 😀